Kernel/Syscalls/select.cpp

```diff
@@ -139,27 +139,29 @@ int Process::sys$select(const Syscall::SC_select_params* params)
     return marked_fd_count;
 }
 
-int Process::sys$poll(const Syscall::SC_poll_params* params)
+int Process::sys$poll(Userspace<const Syscall::SC_poll_params*> user_params)
 {
     REQUIRE_PROMISE(stdio);
+
     // FIXME: Return -EINVAL if timeout is invalid.
-    if (!validate_read_typed(params))
+    Syscall::SC_poll_params params;
+    if (!validate_read_and_copy_typed(&params, user_params))
         return -EFAULT;
 
     SmapDisabler disabler;
 
-    pollfd* fds = params->fds;
-    unsigned nfds = params->nfds;
-    const timespec* timeout = params->timeout;
-    const sigset_t* sigmask = params->sigmask;
+    pollfd* fds = params.fds;
+    unsigned nfds = params.nfds;
 
     if (fds && !validate_read_typed(fds, nfds))
         return -EFAULT;
-    if (timeout && !validate_read_typed(timeout))
-        return -EFAULT;
-    if (sigmask && !validate_read_typed(sigmask))
+
+    timespec timeout = {};
+    if (params.timeout && !validate_read_and_copy_typed(&timeout, params.timeout))
         return -EFAULT;
-    if (!validate_read_typed(fds))
```

Comment:
This appeared to be a duplicate check so I removed? Am I missing something?

Reply:
Thoughts?

Reply:
Sounds good. Also this check was too narrow as it would only validate the first entry in `fds`.